### PR TITLE
Hotfix: Disable APC found_posts filters while running cron

### PR DESCRIPTION
**Pull request by kevinfodness**
Under certain circumstances, the Advanced Post Cache filters for `found_posts` can return the incorrect number of found posts for the Searchpress index query, causing it to prematurely abort. This PR unhooks the APC `found_posts` filters while cron is running an indexing operation.

### CODE REVIEW

File: lib/class-sp-sync-manager.php [hunk 1]
```diff
@@ -166,6 +166,26 @@ public function get_posts( $args = array() ) {
...
+					'found_posts',
+					array( $advanced_post_cache_object, 'found_posts' )
+				);
+				remove_filter(
+					'found_posts_query',
+					array( $advanced_post_cache_object, 'found_posts_query' )
+				);
+			}
+		}
+
```
Comment (mboynes) [inline lib/class-sp-sync-manager.php]: This feels out of scope for this function. I think it would be better in another file, (perhaps a `compat.php` file), and we add an action here to hook into, e.g. `sp_pre_index_loop` or similar.

File: lib/class-sp-sync-manager.php [hunk 2]
```diff
@@ -166,6 +166,26 @@ public function get_posts( $args = array() ) {
 	 * @return bool
 	 */
 	public function do_index_loop() {
+
+		/**
```
Comment (mboynes) [inline lib/class-sp-sync-manager.php]: Nitpicky, but IIRC a multiline comment should only start with two asterisks if it is explicitly a docblock.

File: lib/class-sp-sync-manager.php (hunk 1 above)
Comment (kevinfodness) [reply lib/class-sp-sync-manager.php]: Sure thing!

**Comment by kevinfodness** [comment]
@mboynes this is ready for another look.

File: lib/class-sp-sync-manager.php (hunk 2 above)
Comment (pattiereaves) [reply lib/class-sp-sync-manager.php]: :til:
